src/live_restore/live_restore_server.c

```diff
@@ -0,0 +1,293 @@
+    }
+    __wt_spin_unlock(session, &server->queue_lock);
+
+    return (0);
+}
+
+/*
+ * __live_restore_work_queue_drain --
+ *     Drain the work queue of any remaining items. This function does not need to take the queue
+ *     lock but we do anyway because the semantic is if you touch the queue after it is initialized,
+ *     you hold the lock.
+ */
+static void
+__live_restore_work_queue_drain(WT_SESSION_IMPL *session)
+{
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+    __wt_spin_lock(session, &server->queue_lock);
+    if (!TAILQ_EMPTY(&server->work_queue)) {
+        WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL, *work_item_tmp = NULL;
+        TAILQ_FOREACH_SAFE(work_item, &server->work_queue, q, work_item_tmp)
+        {
+            TAILQ_REMOVE(&server->work_queue, work_item, q);
+            __wt_free(session, work_item->uri);
+            __wt_free(session, work_item);
+        }
+    }
+    WT_ASSERT(session, TAILQ_EMPTY(&server->work_queue));
+    __wt_spin_unlock(session, &server->queue_lock);
+}
+
+/*
+ * __live_restore_worker_run --
+ *     Entry function for a live restore thread. This is called repeatedly from the thread group
+ *     code so it does not need to loop itself.
+ */
+static int
+__live_restore_worker_run(WT_SESSION_IMPL *session, WT_THREAD *ctx)
+{
+    WT_DECL_RET;
+    WT_UNUSED(ctx);
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+
+    __wt_spin_lock(session, &server->queue_lock);
+    if (TAILQ_EMPTY(&server->work_queue)) {
+        /* Stop our thread from running. This will call the stop_func and trigger state cleanup. */
+        F_CLR(ctx, WT_THREAD_RUN);
+        __wt_verbose_debug2(session, WT_VERB_FILEOPS, "%s", "Live restore worker terminating");
+        __wt_spin_unlock(session, &server->queue_lock);
+        return (0);
+    }
+
+    WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
+
+    work_item = TAILQ_FIRST(&server->work_queue);
+    WT_ASSERT(session, work_item != NULL);
+    TAILQ_REMOVE(&server->work_queue, work_item, q);
+    WT_STAT_CONN_SET(session, live_restore_queue_length, --server->queue_size);
+    __wt_verbose_debug2(
+      session, WT_VERB_FILEOPS, "Live restore worker taking queue item: %s", work_item->uri);
+    __wt_spin_unlock(session, &server->queue_lock);
+
+    WT_CURSOR *cursor;
+    WT_SESSION *wt_session = (WT_SESSION *)session;
+
+    /*
+     * Open a cursor so no one can get exclusive access on the object. This prevents concurrent
+     * schema operations like drop and rename. Even if this object is a log file it can have a
+     * cursor opened it. Opening a cursor on a log will prevent it from getting archived.
+     *
+     * If the file no longer exists, which for logs means they could have been archived and for
+     * regular files dropped, don't error out.
+     */
+    WT_RET_ERROR_OK(
+      wt_session->open_cursor(wt_session, work_item->uri, NULL, NULL, &cursor), ENOENT);
+
```

Comment:
55fb2566c

Reply:
Can we keep the comment in the update code? We can say something like "The file existed on connection open when added to the work queue but has since been deleted. Nothing to migrate" if the previous comment was too redundant